test/spmd/test_xla_virtual_device.py

```diff
@@ -88,6 +88,22 @@ def test_non_tensor_scalar(self):
     xt2 = xt1 / 0.5
     torch.allclose(xt2.cpu(), xt1.cpu() / 0.5)
 
+  def test_mark_step_on_virtual_device(self):
+    xm.mark_step()
+    sharding_spec = xs.ShardingSpec(self._get_mesh((1, self.n_devices)), (0, 1))
+    # TODO(JackCaoG)currently, execution will only happen if there is at least one
+    # tensor on non-spmd:0 device.
+    t1 = torch.randn(3, 3, device=xm.xla_device())
```

Comment:
What's this t1 about?

Reply:
so.. her eis another weird bug.. if both `xt1` and `xt2` on `spmd:0`, the `mark_step` will see that there is no tensor on `TPU:0` device and it will skip the execution.. I think it is easier for me to rewrite whole virtual device instead of fixing these bugs to suit both `spmd:0` and `TPU:0` exist at the same time.

Reply:
Good luck... I wonder how was I able to do GPT-2 experiments all the time...

Reply:
I am surprised that code runs at all lol.